tests/functional/test_sts.py

```diff
@@ -23,9 +23,6 @@ class TestSTSPresignedUrl(BaseSessionTest):
     def setUp(self):
         super(TestSTSPresignedUrl, self).setUp()
         self.client = self.session.create_client('sts', 'us-west-2')
-        # Makes sure that no requests will go through
```

Comment:
This was removed because the stubber hooks into the parameter building related event for validating API/parameter calls, but since we did not previously add any expected API calls in test, the stubber will now complain about an unexpected API call.

It seems strange that a stubber would be used for presigned URL's in the first place, but I could see it potentially affect downstream tests though if someone used a stubber to stub out an API call and used the same stubbed client to generate a presigned url in their tests. What people's thoughts are on this?

Reply:
Is there another even the stubber could rely on?

Reply:
Yeah I'd prefer if we could preserve this behavior somehow.  The fact that we're changing a functional test suggests it's possible others will be impacted, especially given the stubber is a public API.

What are the other options here?

Reply:
@joguSD don't think there is another event the stubber could rely on. This PR is having the presigners emit`provide-client-params` and `before-parameter-build` which are the events that would contain the unserialized parameters. Any of the other events down the stack use the serialized version of the parameters. So if we were to hook into another event, we would need to create a new one which we would not want to do.

@jamesls One idea initially that I have is to inject a value into the `context` object a flag like `is_presign_request` in the presigner logic and then in the Stubber class we can check for this flag and pass over asserting the request if the presign flag is present in the context. What's your thoughts on this? Not sure if there really any better options given the stubber currently does not have a way to disambiguate between a presign request and a normal client request.

Reply:
Sorry didn't see this until now.  I'm not a huge fan on this because it unnecessarily couples the stubber and signers.  Ideally the stubber shouldn't care about (or know) whether a request is presigned or not, it would just have better events it could hook into.

However, I'm not sure it's worth the trouble at this point to introduce new events so I think this is fine for now.